### PR TITLE
fix(parser): disallow type parameters on class constructors

### PR DESCRIPTION
**Pull request by srijan-paul**
TypeScript does not allow type parameters on constructors:

```ts
class A {
  constructor<T>(t: T) {} // <- should be an error
}
```

Here's a comparison snippet on:
1. [TypeScript playground](https://www.typescriptlang.org/play/?#code/MYGwhgzhAECC0G8BQ1XWAewHYQC4CcBXYXDfAHlgD4AKASkSRTQF8kWg) (syntax error).
2. [SWC playground](https://play.swc.rs/?version=1.10.1&code=H4sIAAAAAAAAA0vOSSwuVnBWqOZSUEjOzysuKSpNLskvsgmx09BUqFao5aoFAOf92%2FAiAAAA&config=H4sIAAAAAAAAA1WPSw7DIAwF9zkF8rrbdtE79BCIOhERP9mOVBTl7oUE0maH3xszsA5KwcwGnmotxzIkTYx0ziXhHER%2FSgKSE7IhmwRuvRWu1agd4x5tRwOiaUKpW8j3hoOLkbHjLfM22DH%2FC030iZD5ClZUh8nhVTc0Jfj4XvayfaQ%2B9tA%2F4Ad12XkxWH71TaEFh%2B0LYuVI0xQBAAA%3D) (syntax error)
3. [OXC playground](https://playground.oxc.rs/#eNo9T8EKwjAM/ZWSk8IOKngpInjxCzx66Wo2Cl0zkk4do/9u6+guyXshL+9lAQsarDci6qaWZ1DKUpDIk43El8d1t1+nSpWaniFBAwR6AZ5CaTKHaL6gswQb8C7EisXSiBuZh5Z8ZZFNkI54AN0ZL5gaGA0Lcr6YcblS8bb6H0A03GO2AJTT4XiGvGHphT2WMJkMLrjOVXH+JTL5u6dPEb+RW5KcaTVN6QerbleM) (parses as valid syntax)

This PR makes OXC behave exactly like `tsc`, with a little test case.
Happy holidays!

### CODE REVIEW

**Comment by Dunqing** [comment]
Run `just coverage` to update conformance snapshots

File: crates/oxc_parser/src/lib.rs [hunk 1]
```diff
@@ -637,6 +637,19 @@ mod test {
...
+        let source_type = SourceType::from_path(Path::new("class.ts")).unwrap();
+        let source = "class A { constructor<T>() {} }\n";
+        let ret = Parser::new(&allocator, source, source_type).parse();
+        assert_eq!(ret.errors.len(), 1);
+        assert_eq!(
+            ret.errors.first().unwrap().to_string(),
+            "Type parameters cannot appear on a constructor declaration"
+        );
+    }
+
```
Comment (Dunqing) [inline crates/oxc_parser/src/lib.rs:652]: Conformance test has caught this, so we don't need to add test for it

**Review by Dunqing — DISMISSED** [review]

File: crates/oxc_parser/src/lib.rs (hunk 1 above)
Comment (srijan-paul) [reply crates/oxc_parser/src/lib.rs]: Ah, right.

**Comment by srijan-paul** [comment]
> Run just coverage to update conformance snapshots
thanks, done.

**Pull request merged**